### DISCUSSION TITLE
Database unit in gdsfactory

### DISCUSSION THREAD

**Discussion by manoranjanminz** [gdsfactory/gdsfactory#2189] [Q&A] [answered]
By default, the database unit is 1nm. How can we change the database unit to some other value in gdsfactory?

**Comment by joamatab** [comment, marked as answer]
```python
    import gdsfactory as gf 
    c = gf.components.mzi()
    gdspath =c.write_gds(precision=5e-9)
    gf.show(gdspath)
```

**Reply by manoranjanminz** [reply]
Thanks Joaquin, that worked.

**Reply by scomb14** [reply]
I tried to set both unit and precision while exporting to gds but these options now seem deprecated.
When runnig the code above I get: 'precision is deprecated'.
How can these parameters be changed in new versions?

**Reply by joamatab** [reply]
You can easily change it now, for example if you want to set it up to 5nm with

`gf.kcl.dbu = 5e-3`

**Reply by joamatab** [reply]
i will add some docs here

https://gdsfactory.github.io/gdsfactory/notebooks/12_grid_snapping.html#

**Reply by scomb14** [reply]
It now works perfectly.
Thank you very much for your help.

**Comment by stormragey** [comment]
What if I want the precision to be 0.1nm?

**Reply by sebastian-goeldi** [reply]
This is a very old discussion, in the new gdsfactory you set it relative to 1 um.

`gf.kcl.dbu=0.0001`

Note that in some places 1nm might be hardcoded.

**Reply by stormragey** [reply]
Thanks for the reply, I found this and tried before but it never works on my side when setting the gf.kcl.dbu=0.0001
 A exmaple code as shown below:
-------------------------------------------
import gdsfactory as gf
gf.kcl.dbu=0.0001
gf.clear_cache()
cell = gf.Component("Test")
rect = gf.components.rectangle(size=(0.1234, 0.4567), layer=(6, 0))

----------------------------------------------
This always brings up a value Error message"Port width needs to be even to snap to grid properly and greater than 0. 1 DBU is 0.0001 um. Port width must be a multiple of 0.0002 um.", it only works when gf.kcl.dbu=0.001.
When I google online, it was said kcl cannot set precision below 1nm, I am not sure whether this is real reason.

**Reply by sebastian-goeldi** [reply]
The reason is the `0.4567` size. It's not dividable by 2 dbu. Rectangles have ports and MUST be dividable by 2 dbu. Your y-size is 4567 dbu big. Round up or down to make it happen. Otherwise I'd suggest to use polygons directly if you need another size

But you can already see the dbu being set to 0.1nm as the error complains

**Reply by stormragey** [reply]
Hello, Sebastian,
Thanks for the suggestion. The polygon would work for me even with kcl.dbu=1e-4.

Just want to clarify here in case others see this post that I do not think it's the dbu issue leading to error message.
Even changing the dbu to be 1e-5, **0.4567** still cannot be drawing in the rectangle shape due to port error. (I guess gf.components.rectangle having port setting inside which cannot having dimension < 1nm)
This port error won't exsit in the polygon.

**Reply by sebastian-goeldi** [reply]
it's the snap_to_grid that is causing you grievance here. It snaps to `nm` not `dbu`. See in `gf.components.compass` where this happens.

cc @joamatab
